selector.go

```diff
@@ -22,8 +26,16 @@ type IDSelectorBatch struct {
 
 // Delete frees the memory associated with s.
 func (s *IDSelectorBatch) Delete() {
-	C.faiss_IDSelector_free(s.sel)
-	C.faiss_IDSelector_free(s.batchSel)
+	if s == nil {
+		return
+	}
+
+	if s.sel != nil {
+		C.faiss_IDSelector_free(s.sel)
+	}
+	if s.batchSel != nil {
```

Comment:
This check and the previous one - are they necessary? free on nil will be a no-op.